examples/src/main/java/mil/nga/giat/geowave/examples/query/SpatialQueryExample.java

```diff
@@ -0,0 +1,497 @@
+package mil.nga.giat.geowave.examples.query;
+
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.geom.Point;
+import com.vividsolutions.jts.io.ParseException;
+import com.vividsolutions.jts.io.WKTReader;
+import mil.nga.giat.geowave.adapter.vector.FeatureDataAdapter;
+import mil.nga.giat.geowave.adapter.vector.VectorDataStore;
+import mil.nga.giat.geowave.core.geotime.GeometryUtils;
+import mil.nga.giat.geowave.core.geotime.IndexType;
+import mil.nga.giat.geowave.core.geotime.store.query.SpatialQuery;
+import mil.nga.giat.geowave.core.index.ByteArrayId;
+import mil.nga.giat.geowave.core.store.CloseableIterator;
+import mil.nga.giat.geowave.datastore.accumulo.BasicAccumuloOperations;
+import mil.nga.giat.geowave.datastore.accumulo.metadata.AccumuloAdapterStore;
+import org.apache.accumulo.core.client.AccumuloException;
+import org.apache.accumulo.core.client.AccumuloSecurityException;
+import org.apache.accumulo.core.client.Connector;
+import org.apache.accumulo.core.client.mock.MockInstance;
+import org.apache.accumulo.core.client.security.tokens.PasswordToken;
+import org.apache.log4j.Logger;
+import org.geotools.feature.AttributeTypeBuilder;
+import org.geotools.feature.simple.SimpleFeatureBuilder;
+import org.geotools.feature.simple.SimpleFeatureTypeBuilder;
+import org.geotools.filter.text.cql2.CQL;
+import org.geotools.filter.text.cql2.CQLException;
+import org.geotools.geometry.jts.JTSFactoryFinder;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+
+import java.util.ArrayList;
+
+/**
+ * This class is intended to provide a few examples on running Geowave queries of different types:
+ * 1- Querying by polygon a set of points.
+ * 2- Filtering on attributes of features using CQL queries
+ * 3- Ingesting polygons, and running polygon intersect queries.
+ * You can check all points, geometries and query accuracy in a more visual manner @ http://geojson.io/
+ */
+public class SpatialQueryExample {
+    private static Logger log = Logger.getLogger(SpatialQueryExample.class);
+
+	//We'll use GeoWave's VectorDataStore, which allows to run CQL rich queries
+	private static VectorDataStore dataStore;
```

Comment:
to further clarify - vectordatastore provides some query methods that are specific to SimpleFeature data, such as the CQL query on line 370 of this example.  In 0.9 we try to eliminate the need for this specialized interface, the query method that you can use for CQL filtering is the same query method as in the DataStore interface, you just pass in a query object that is a CQLQuery within the vector adapter module. I'm happy with accepting this as the right example for CQL queries under 0.9-snapshot, and updating it for 0.9 as we pull in some of the broader changes (so our GEOWAVE-238-50_merge branch is actually where a lot of development is being staged for some 0.9 API changes)